### PR TITLE
Update phone location

### PR DESCRIPTION
**Pull request by gbova**
FB: https://manage.dimagi.com/default.asp?269773

The first commit fetches and removes all user IDs from the location that is being deleted.

The second commit updates UserFixtureType.LOCATION to ensure the change is reflected in locations fixture on the next sync.

### CODE REVIEW

**Comment by nickpell** [comment]
Test failure looks real:

```
FAIL: corehq.apps.users.tests.test_location_assignment:CCUserLocationAssignmentTest.test_deleting_location_updates_user
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/users/tests/test_location_assignment.py", line 104, in test_deleting_location_updates_user
    self.assertAssignedLocations([])
  File "/mnt/commcare-hq-ro/corehq/apps/users/tests/test_location_assignment.py", line 130, in assertAssignedLocations
    self.assertListEqual(user.assigned_location_ids, expected_location_ids)
AssertionError: Lists differ: [u'5b9f665cdd9a419c9b5d6d89811... != []
First list contains 2 additional elements.
First extra element 0:
u'5b9f665cdd9a419c9b5d6d89811f45dc'
- [u'5b9f665cdd9a419c9b5d6d89811f45dc', u'68ba2c203b624a8e9bb7566925e8a040']
+ []
```

**Comment by emord** [comment]
@gbova I think I fixed the tests in the latest commit. Can go over this today.

Also could you add a test to this file https://github.com/dimagi/commcare-hq/blob/update-phone-location/corehq/apps/locations/tests/test_location_fixtures.py#L538

I think it should be something like the WebUser test linked, but instead of creating a web user it should create a CommCareUser and you should test deleting locations. (I don't think that's tested in that file at the moment)

File: corehq/apps/locations/tests/test_location_fixtures.py [hunk 1]
```diff
@@ -711,13 +712,42 @@ def test_archiving_location_should_resync(self):
...
+        location.save()
+        after_save = datetime.utcnow()
+        self.assertEqual('Riverrun', location.name)
+        locations_queryset = SQLLocation.objects.filter(pk=location.pk)
+
+        # Should not resync if last sync was after location was saved
+        self.assertFalse(
+            should_sync_locations(SyncLog(date=after_save), locations_queryset, self.user.to_ota_restore_user())
+        )
+
```
Comment (emord) [inline corehq/apps/locations/tests/test_location_fixtures.py]:
```
self.user.assign_to_location 
after_assignment = datetime.utcnow()

assertTrue should_sync after_save
assertFalse should_sync after_assignment
```

File: corehq/apps/locations/tests/test_location_fixtures.py [hunk 2]
```diff
@@ -711,13 +712,42 @@ def test_archiving_location_should_resync(self):
...
+        # Delete the location
+        location.full_delete()
+        after_delete = datetime.utcnow()
+
+        # TODO: Should resync if last sync was after location was saved but before location was deleted
```
Comment (emord) [inline corehq/apps/locations/tests/test_location_fixtures.py]:
```
assertTrue should_sync after_assignment
assertFalse should_sync after_delete
```

**Review by emord — COMMENTED** [review]
@gbova Thanks for uploading this test. I made a couple of comments with psuedo code for what I think the test should test. Let me know if that makes sense

**Pull request merged**